src/igvfd/audit/file_set.py

```diff
@@ -114,17 +119,32 @@ def audit_missing_seqspec(value, system):
                 if not sequence_file_object.get('seqspecs'):
                     no_seqspec.append(file)
         if no_seqspec:
-            no_seqspec = ', '.join([audit_link(path_to_text(file_no_seqspec), file_no_seqspec)
-                                   for file_no_seqspec in no_seqspec])
+            # Audit 1: Single cell SeqFiles are expected linked to ConfigFile seqspecs
             if single_cell_check(system, value, object_type):
                 audit_message = get_audit_message(audit_missing_seqspec, index=1)
+                no_seqspec_paths = join_obj_paths(data_object_paths=no_seqspec)
+                detail = (
+                    f'{object_type} {audit_link(path_to_text(value["@id"]), value["@id"])} has sequence file(s): '
+                    f'{no_seqspec_paths} which do not have any `seqspecs`.'
+                )
+                yield AuditFailure(audit_message.get('audit_category', ''), f'{detail} {audit_message.get("audit_description", "")}', level=audit_message.get('audit_level', ''))
+            # Audit 2: Check if non-single cell SeqFiles without seqspec ConfigFile still have seqspec_document.
             else:
-                audit_message = get_audit_message(audit_missing_seqspec, index=0)
-            detail = (
-                f'{object_type} {audit_link(path_to_text(value["@id"]), value["@id"])} has sequence file(s): '
-                f'{no_seqspec} which do not have any `seqspecs`.'
-            )
-            yield AuditFailure(audit_message.get('audit_category', ''), f'{detail} {audit_message.get("audit_description", "")}', level=audit_message.get('audit_level', ''))
+                no_seqspec_doc = []
+                for seqfile in no_seqspec:
+                    file_obj = system.get('request').embed(seqfile, '@@object?skip_calculated=true')
```

Comment:
you're requesting the same sequence file object twice, any way you could avoid this?

Reply:
referring to line 116